File: OBJCSample/OBJCSample/classes/AZHijriCalendar.m
//
//  AZHijriCalendar.m
//  AZCalendarOBJC
//
//  Created by Ali on 4/14/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

#import "AZHijriCalendar.h"
#import "DateType.h"
@implementation AZHijriCalendar


+ (DateType*) getHijriDate {
    DateType *result = [[DateType alloc] init];
    
    NSCalendar *gregorianCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];

    
     NSDateComponents *gregorianComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:[[NSDate alloc] initWithTimeIntervalSinceNow:0]];

    NSDate *date = [gregorianCalendar dateFromComponents:gregorianComponents];

    
    NSCalendar *hijriCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierIslamicCivil];
    
   
    NSDateComponents *hijriComponent = [hijriCalendar components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:date];

    result.day = hijriComponent.day;
     result.month = hijriComponent.month;
     result.year = hijriComponent.year;
    
    
    
    return  result;
}


+ (NSInteger) getHijriDayNumber{

    return  [[AZHijriCalendar getHijriDate] day];
}
+ (NSInteger) getHijriMonthNumber{
    
     return  [[AZHijriCalendar getHijriDate] month];
}
+ (NSInteger) getHijriYearNumber{
    return  [[AZHijriCalendar getHijriDate] year];
}


@end

File: SwiftSample/SwiftSample/classes/AZHijriCalendar.swift
//
//  AZHijriCalendar.swift
//  AZCalendarSwift
//
//  Created by Ali on 4/11/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

import Foundation
 public class AZHijriCalendar {
    
    
    static func getHijriDayNumber()->Int{
        
        return AZHijriCalendar.getHijriDate().day
    }
    static func getHijriMonthNumber()->Int{
        return AZHijriCalendar.getHijriDate().month
    }
    static func getHijriYearNumber()->Int{
        return AZHijriCalendar.getHijriDate().year
    }
    
    
   private static func getHijriDate() -> (day : Int , month : Int , year : Int) {
    
    
        let gregorianCalendar = Calendar.init(identifier: .gregorian)
        let gregorianComponents = Calendar.current.dateComponents([.year, .month, .day], from: Date.init(timeIntervalSinceNow: 0))
        
        
        let date = gregorianCalendar.date(from: gregorianComponents)
       
        
        let hijriCalendar = Calendar.init(identifier: .islamicCivil)
        let hijriComponents = hijriCalendar.dateComponents([.year, .month, .day], from: date!)
     
        return (day : hijriComponents.day! , month : hijriComponents.month! , year : hijriComponents.year!)
    
    }
    
}
